packages/rocketchat-ui/client/views/app/room.js

```diff
@@ -730,6 +729,17 @@ Template.room.events({
 				}
 
 				FlowRouter.goToRoomById(room._id);
```

Comment:
``goToRoomByID`` does not resolve when there is no subscription exists for the user. The user is not a member of a group and he/she not allowed to navigate to the target channel(Both thread and mention-link).

Reply:
Exactly. This is why I implemented the else tarting from line 732.
The room in line 726 will only be assigned also if a subscription exists for the user.
Can you tell me which navigation does not work?

Reply:
User created a new thread from 'General' channel. Later he was removed as owner and kicked out of the room.Now when the user tries again to access the link. it works for the first time but not the subsequent clicks.

Reply:
Navigation works as excepted.
What is tested
1) Public Channel:
   - Navigation between public should be possible even-though the user is not member of the target channel
2) Private Channel:
    - Navigation between private should be not possible if the user is not member of the target channel. 

What is missing: 
Show the **Not Authorised** Page if the user do not have access to the target channel.